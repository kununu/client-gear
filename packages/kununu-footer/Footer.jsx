import React, {PropTypes} from 'react';
import Logo from 'kununu-logo';

import styles from './index.scss';

export default function Footer ({
  children,
  infoText,
  tuv,
}) {
  return (
    <footer
      role="banner"
      id="footer"
      className={`navbar-default ${styles.footer}`}
    >
      <div className="container-fluid">
        <div className={`row ${styles.flex}`}>
          {children.map((child, index) => {
            if (child.props.type !== 'col') return false;
            return (
              <div className={`${styles.menuColumns} ${child.props.menuClass}`} key={index}>
                {child}
              </div>
            );
          })}

          {tuv ?
            <div className={`${styles.tuvColumn} no-padding hidden-xs hidden-sm`}>
              <img
                alt={tuv.alt}
                className={styles.tuv}
                data-toggle="tooltip"
                data-placement="top"
                data-html="true"
                src={tuv.src}
                title={tuv.title}
              />
            </div>
            : ''
          }

          <div className={`${styles.infoTextColumn} text-right text-center-xs text-center-sm text-muted`}>
            <Logo shade="light" />

            <p className={styles.infoText}>
              {infoText}
            </p>

            <p className="text-xs">
              made with <i className="fa fa-heart-o text-red" /> in Vienna
            </p>
          </div>
        </div>
        <div className={styles.bottomMenu}>
          <div>
            {children.filter((child) => child.props.type === 'row')}
          </div>
          <div className="hidden-xs">
            {children.filter((child) => child.type.name === 'Dropdown')}
          </div>
        </div>
      </div>
    </footer>
  );
}

Footer.propTypes = {
  children: PropTypes.arrayOf(PropTypes.object),
  infoText: PropTypes.element,
  tuv: PropTypes.shape({
    alt: React.PropTypes.string,
    src: React.PropTypes.string,
    title: React.PropTypes.string,
  }),
};
