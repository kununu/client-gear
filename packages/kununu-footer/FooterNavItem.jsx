import React, {PropTypes} from 'react';

import styles from './index.scss';

export default function FooterNavItem ({
  active,
  children,
}) {
  return (
    <li className={`${styles.footerNavItem} ${active && styles.active}`}>
      {children}
    </li>
  );
}

FooterNavItem.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.element,
};
