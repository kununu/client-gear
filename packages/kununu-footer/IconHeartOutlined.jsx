import React from 'react';
import PropTypes from 'prop-types';

const IconHeartOutlined = ({className}) => (
  <svg
    width="37px"
    height="31px"
    viewBox="0 0 37 31"
    version="1.1"
    className={className}
  >
    <title>Heart</title>
    <path
      d="M34,9.40178571 C34,8.31695886 33.8560282,7.35937915 33.5680804,6.52901786 C33.2801325,5.69865656 32.9118326,5.03906494 32.4631696,4.55022321 C32.0145067,4.06138148 31.4687532,3.66294797 30.8258929,3.35491071 C30.1830325,3.04687346 29.5535745,2.83928625 28.9375,2.73214286 C28.3214255,2.62499946 27.6651821,2.57142857 26.96875,2.57142857 C26.2723179,2.57142857 25.5223254,2.74218579 24.71875,3.08370536 C23.9151746,3.42522492 23.1752266,3.85379206 22.4988839,4.36941964 C21.8225413,4.88504722 21.243306,5.36718526 20.7611607,5.81584821 C20.2790154,6.26451117 19.8772337,6.67633741 19.5558036,7.05133929 C19.3147309,7.34598362 18.9866092,7.49330357 18.5714286,7.49330357 C18.1562479,7.49330357 17.8281262,7.34598362 17.5870536,7.05133929 C17.2656234,6.67633741 16.8638417,6.26451117 16.3816964,5.81584821 C15.8995512,5.36718526 15.3203159,4.88504722 14.6439732,4.36941964 C13.9676305,3.85379206 13.2276826,3.42522492 12.4241071,3.08370536 C11.6205317,2.74218579 10.8705392,2.57142857 10.1741071,2.57142857 C9.47767509,2.57142857 8.82143165,2.62499946 8.20535714,2.73214286 C7.58928263,2.83928625 6.95982464,3.04687346 6.31696429,3.35491071 C5.67410393,3.66294797 5.12835046,4.06138148 4.6796875,4.55022321 C4.23102454,5.03906494 3.86272465,5.69865656 3.57477679,6.52901786 C3.28682892,7.35937915 3.14285714,8.31695886 3.14285714,9.40178571 C3.14285714,11.651797 4.39507676,14.0290053 6.89955357,16.5334821 L18.5714286,27.7834821 L30.2232143,16.5535714 C32.741084,14.0357017 34,11.651797 34,9.40178571 L34,9.40178571 Z M36.5714286,9.40178571 C36.5714286,12.3616219 35.0379618,15.3749847 31.9709821,18.4419643 L19.4553571,30.4955357 C19.2142845,30.7366083 18.9196446,30.8571429 18.5714286,30.8571429 C18.2232125,30.8571429 17.9285726,30.7366083 17.6875,30.4955357 L5.15178571,18.4017857 C5.01785647,18.2946423 4.83370653,18.1205369 4.59933036,17.8794643 C4.36495419,17.6383917 3.99330612,17.19978 3.484375,16.5636161 C2.97544388,15.9274522 2.52009129,15.2745569 2.11830357,14.6049107 C1.71651585,13.9352645 1.3582605,13.1250048 1.04352679,12.1741071 C0.728793069,11.2232095 0.571428571,10.2991116 0.571428571,9.40178571 C0.571428571,6.45534241 1.4218665,4.15179402 3.12276786,2.49107143 C4.82366922,0.830348839 7.17409214,0 10.1741071,0 C11.0044684,0 11.8515582,0.143971775 12.7154018,0.431919643 C13.5792454,0.719867511 14.3828088,1.10825648 15.1261161,1.59709821 C15.8694234,2.08593994 16.5089259,2.54464071 17.0446429,2.97321429 C17.5803598,3.40178786 18.0892833,3.85714045 18.5714286,4.33928571 C19.0535738,3.85714045 19.5624973,3.40178786 20.0982143,2.97321429 C20.6339312,2.54464071 21.2734338,2.08593994 22.0167411,1.59709821 C22.7600484,1.10825648 23.5636118,0.719867511 24.4274554,0.431919643 C25.291299,0.143971775 26.1383887,0 26.96875,0 C29.968765,0 32.3191879,0.830348839 34.0200893,2.49107143 C35.7209906,4.15179402 36.5714286,6.45534241 36.5714286,9.40178571 L36.5714286,9.40178571 Z"
    />
  </svg>
);

IconHeartOutlined.propTypes = {
  className: PropTypes.string.isRequired,
};

export default IconHeartOutlined;
