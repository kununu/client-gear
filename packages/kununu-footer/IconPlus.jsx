import React from 'react';
import PropTypes from 'prop-types';

const IconPlus = ({className}) => (
  <svg
    width="29px"
    height="29px"
    className={className}
    viewBox="0 0 29 29"
    version="1.1"
  >
    <title>Plus</title>
    <path
      d="M28.2857143,12.2142857 L28.2857143,16.0714286 C28.2857143,16.6071455 28.0982162,17.0624981 27.7232143,17.4375 C27.3482124,17.8125019 26.8928598,18 26.3571429,18 L18,18 L18,26.3571429 C18,26.8928598 17.8125019,27.3482124 17.4375,27.7232143 C17.0624981,28.0982162 16.6071455,28.2857143 16.0714286,28.2857143 L12.2142857,28.2857143 C11.6785688,28.2857143 11.2232162,28.0982162 10.8482143,27.7232143 C10.4732124,27.3482124 10.2857143,26.8928598 10.2857143,26.3571429 L10.2857143,18 L1.92857143,18 C1.39285446,18 0.937501875,17.8125019 0.5625,17.4375 C0.187498125,17.0624981 0,16.6071455 0,16.0714286 L0,12.2142857 C0,11.6785687 0.187498125,11.2232162 0.5625,10.8482143 C0.937501875,10.4732124 1.39285446,10.2857143 1.92857143,10.2857143 L10.2857143,10.2857143 L10.2857143,1.92857143 C10.2857143,1.39285446 10.4732124,0.937501875 10.8482143,0.5625 C11.2232162,0.187498125 11.6785688,0 12.2142857,0 L16.0714286,0 C16.6071455,0 17.0624981,0.187498125 17.4375,0.5625 C17.8125019,0.937501875 18,1.39285446 18,1.92857143 L18,10.2857143 L26.3571429,10.2857143 C26.8928598,10.2857143 27.3482124,10.4732124 27.7232143,10.8482143 C28.0982162,11.2232162 28.2857143,11.6785687 28.2857143,12.2142857 L28.2857143,12.2142857 Z"
    />
  </svg>
);

IconPlus.propTypes = {
  className: PropTypes.string.isRequired,
};

export default IconPlus;
