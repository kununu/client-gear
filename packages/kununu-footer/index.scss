@import '~kununu-theme/scss/base/variables';

.footer {
  display: table;
  padding-top: 20px;
  margin-bottom: 0;
  width: 100%;
}

.footerNav {
  list-style: none;
  margin-bottom: 0;
  overflow: hidden;
  padding-left: 0;
  transition: all .5s;

  a {
    color: $text-muted-dark;
    display: inline-block;
    font-size: $font-size-xs;
    width: 100%;
    padding: 4px 0;

    @media (min-width: $screen-sm-min) {
      font-size: $font-size-xs;
    }

    @media (max-width: $screen-xs-max) {
      display: block;
      transition: padding .1s cubic-bezier(.215, .61, .355, 1) .5s;
      font-size: $font-size-base;
    }

    &:hover {
      text-decoration: underline;
      color: $white;
    }
  }

  @media (max-width: $screen-xs-max) {
    max-height: 0;

    &.open {
      max-height: 220px;

      a {
        transition: padding .4s cubic-bezier(.215, .61, .355, 1);
        padding-left: 20px;
      }
    }
  }

  &.row {
    padding: 0;
    margin: 0;

    li {
      display: inline-block;
      padding-left: 2px;

      &,
      a {
        display: inline-block;
      }

      a {
        padding: 4px 0;
        font-size: $font-size-small;

        &:hover {
          text-decoration: none;
          color: $white;
        }
      }

      &:after {
        content: '\2219'; // bullet http://codepoints.net/U+2219
        display: inline-block;
        padding-left: 2px;
      }

      &:last-child {
        padding-left: 0;

        &:after {
          content: none;
        }
      }
    }
  }
}

.tuvColumn,
.menuColumns,
.infoTextColumn {
  float: left;
  position: relative;
  min-height: 1px;
  padding-left: 20px;
  padding-right: 20px;
}

.flex {
  @media (min-width: $screen-md-min) {
    display: flex;
  }
}

.menuColumns,
.tuvColumn,
.infoTextColumn {
  @media (min-width: $screen-md-min) {
    flex-grow: 1;
  }
}

.menuColumns {
  @media (min-width: $screen-md-min) {
    min-width: 200px;
  }

  @media (max-width: $screen-sm-max) {
    width: 25%;
  }

  @media (max-width: $screen-xs-max) {
    width: 100%;
  }
}

.tuvColumn {
  @media (min-width: $screen-md-min) {
    min-width: 120px;
  }

  @media (max-width: $screen-md-max) {
    width: 8.333333%;
  }
}

.infoTextColumn {
  @media (min-width: $screen-md-min) {
    max-width: 550px;
  }

  @media (max-width: $screen-sm-max) {
    width: 100%;
  }
}

div.title {
  margin-top: 0;
}

.footerNavItem {
  position: relative;
  display: block;

  a {
    width: auto;
  }

  &.active a {
    font-weight: $font-weight-semibold;
  }
}



.tuv {
  padding: 0;
  width: 128px;
}

.infoText {
  font-size: $font-size-sm;

  @media (max-width: $screen-sm-max) {
    width: 100%;
  }
}

.bottomMenu {
  border-top: 1px solid $gray-base-60;
  display: table;
  padding-top: 10px;
  width: 100%;

  > div {
    float: right;

    &:first-child {
      float: left;
    }
  }
}

.accordionTitle {
  background-color: transparent;
  border: 0;
  color: $gray-base-40;
  font-weight: $font-weight-semibold;
  margin-top: 18px;
  outline: none;
  position: relative;
  padding: 0 0 11px;
  text-align: left;
  width: 100%;
  transition: all 1s ease;

  .icon {
    position: absolute;
    right: 0;
    transition: all .3s ease;
  }

  &.open {
    .icon {
      transform: rotate(45deg);
    }
  }

  &:hover {
    text-decoration: none;
  }
}

.itemIcon {
  display: none;

  @media (max-width: $screen-xs-max) {
    display: block;
    float: right;
  }
}

.titleIcon {
  display: none;

  @media (max-width: $screen-xs-max) {
    display: inline-block;
    padding-left: 10px;
  }
}
