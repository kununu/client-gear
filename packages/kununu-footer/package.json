{
  "name": "kununu-footer",
  "version": "4.0.0",
  "description": "kununu Footer",
  "main": "dist",
  "author": "kununu",
  "license": "ISC",
  "devDependencies": {
    "babel-cli": "^6.23.0",
    "cross-env": "^3.1.4",
    "kununu-theme": "^3.0.0",
    "node-sass": "^4.5.0",
    "node-sass-module-importer": "^0.1.0",
    "react": "^15.4.2",
    "react-dom": "^15.4.2",
    "rimraf": "^2.6.0",
    "react-bootstrap": "^0.30.7",
    "nukleus": "7.0.0"
  },
  "dependencies": {
    "kununu-logo": "3.0.0"
  },
  "peerDependencies": {
    "kununu-theme": "^3.0.0",
    "nukleus": "7.0.0",
    "react": "^15.4.2",
    "react-dom": "^15.4.2",
    "react-bootstrap": "^0.30.7"
  },
  "scripts": {
    "prepublish": "yarn dist",
    "babel": "babel $(find ./ -name '*.jsx' ! -name '*.test.jsx' ! -ipath '*node_modules*') --out-dir ./dist",
    "sass": "node-sass ./index.scss --output ./dist --importer node_modules/node-sass-module-importer",
    "clean": "rimraf ./dist",
    "copy-imgs": "mkdir dist/img && cp ./img/*.svg ./dist/img",
    "dist": "npm run clean && cross-env NODE_ENV=production yarn babel && yarn sass && yarn copy-imgs"
  }
}
