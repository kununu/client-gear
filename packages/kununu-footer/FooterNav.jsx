import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';

import styles from './index.scss';
import IconPlus from '@kununu/kununu-icons/dist/Plus';

export default class FooterNav extends Component {
  state = {
    open: false,
  }

  onClickButton = () => {
    this.setState({
      open: !this.state.open,
    });
  }

  getMenuTitle = item => (
    <span>
      {item.value}
      {item.icon ?
        <span className={styles.titleIcon}>
          {item.icon}
        </span>
        : ''}
    </span>
  )

  getItem = item => (
    <span>
      {item.value}
      {item.icon ?
        <span className={styles.itemIcon}>{item.icon}</span>
        : ''}
    </span>
  )

  getActiveItem = () => {
    const {items} = this.props;
    const activeItem = items.filter(item => item.active)[0];

    return this.getMenuTitle(activeItem || items[0]);
  }

  isActive = item => item.active;

  render () {
    const {
      dynamicNav,
      items,
      title,
      type,
    } = this.props;

    return (
      <Fragment>
        {title || dynamicNav ?
          <Fragment>
            <span className={styles.title}>
              {title}
            </span>
            <button
              className={`${styles.accordionTitle} ${this.state.open && styles.open}`}
              onClick={this.onClickButton}
            >
              <span>
                {dynamicNav ? this.getActiveItem() : title}
              </span>
              <IconPlus className={`${styles.plus} ${styles.icon}`} />
            </button>
          </Fragment>
          : ''
        }
        <ul
          role="navigation"
          className={`
            ${styles.footerNav}
            ${styles[type]}
            ${this.state.open && styles.open}`}
        >
          {items.map((item, index) => (
            <li
              key={index}
              className={`${styles.footerNavItem} ${this.isActive(item) && styles.active}`}
            >
              {React.cloneElement(item.link, {}, this.getItem(item))}
            </li>
          ))}
        </ul>
      </Fragment>
    );
  }
}

FooterNav.propTypes = {
  dynamicNav: PropTypes.bool,
  items: PropTypes.arrayOf(PropTypes.object),
  title: PropTypes.string,
  type: PropTypes.oneOf(['row', 'col']),
};

FooterNav.defaultProps = {
  type: 'col',
};
