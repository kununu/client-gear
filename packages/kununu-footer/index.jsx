import React, {Component} from 'react';
import PropTypes from 'prop-types';
import ToolTip from 'nukleus/dist/components/ToolTip';
import DropDown from 'nukleus/dist/components/DropDown';
import HeartIcon from '@kununu/kununu-icons/dist/HeartOutline';
import Logo from '@kununu/kununu-logo';

import FooterNav from './FooterNav';
import styles from './index.scss';
import TuvIcon from './Tuv';

export default class Footer extends Component { // eslint-disable-line
  render () {
    const {
      container,
      infoText,
      items,
      pathname,
      simpleMobile,
      tuv,
      assetsPath,
    } = this.props;

    return (
      <footer
        role="banner"
        id="footer"
        className={`${styles.footer} ${simpleMobile ? styles.simpleMobile : ''}`}
      >
        <div className={container}>
          <div className={`${styles.flex} ${styles.contentSection}`}>
            <div className={`${styles.menuColumns} ${styles.visibleXs}`}>
              <FooterNav
                dynamicNav
                items={items.countrySwitcher}
                pathname={pathname}
                type="col"
              />
            </div>
            {items.navs.cols.map((item, index) => (
              <div className={styles.menuColumns} key={index}>
                <FooterNav
                  items={item.items}
                  pathname={pathname}
                  title={item.title}
                  type="col"
                />
              </div>
            ),
            )}
            {tuv ? (
              <div
                className={`
                  ${styles.tuvColumn}
                  ${styles.hiddenSm}
                `}
              >
                <ToolTip
                  icon={<TuvIcon
                    className={styles.tuvIcon}
                    assetsPath={assetsPath}
                  />}
                  label="TUV"
                  content={(
                    <ul className={styles.tooltipContent}>
                      <li>Anonyme Bewertung</li>
                      <li>Schutz der pers√∂nlichen Daten</li>
                      <li>Transparente Bewertungsstandards</li>
                    </ul>
                  )}
                />
              </div>
            ) : null
            }
            <div className={`${styles.infoTextColumn}`}>
              <div className={styles.logoContainer}>
                <Logo assetsPath={assetsPath} />
              </div>

              <p className={styles.infoText}>
                {infoText}
              </p>

              <p className={styles.infoText}>
                made with <HeartIcon className={`${styles.heart} ${styles.icon}`} /> in Vienna, Boston, Porto, Berlin
              </p>
            </div>
          </div>
          <div className={styles.bottomMenu}>
            <div>
              {items.navs.rows.map((item, index) => (
                <FooterNav
                  key={index}
                  pathname={pathname}
                  items={item.items}
                  type="row"
                />
              ),
              )}
            </div>
            <div className={`${styles.dropdown} ${styles.hiddenXs}`}>
              <DropDown
                position="top"
                items={items.countrySwitcher}
              />
            </div>
          </div>
        </div>
      </footer>
    );
  }
}

Footer.propTypes = {
  container: PropTypes.string,
  infoText: PropTypes.element.isRequired,
  assetsPath: PropTypes.string,
  items: PropTypes.shape({
    countrySwitcher: PropTypes.arrayOf(PropTypes.shape({
      active: PropTypes.bool,
      icon: PropTypes.element,
      link: PropTypes.element.isRequired,
      value: PropTypes.string.isRequired,
    })),
    navs: PropTypes.shape({
      cols: PropTypes.arrayOf(PropTypes.shape({
        items: PropTypes.arrayOf(PropTypes.shape({
          link: PropTypes.element.isRequired,
          value: PropTypes.string.isRequired,
        })),
        title: PropTypes.string,
      })),
      rows: PropTypes.arrayOf(PropTypes.shape({
        items: PropTypes.arrayOf(PropTypes.shape({
          link: PropTypes.element.isRequired,
          value: PropTypes.string.isRequired,
        })),
        title: PropTypes.string,
      })),
    }),
  }),
  pathname: PropTypes.string.isRequired,
  simpleMobile: PropTypes.bool,
  tuv: PropTypes.bool,
};

Footer.defaultProps = {
  simpleMobile: false,
  container: 'container-fluid',
  assetsPath: 'https://assets.kununu.com/images/footer',
};
