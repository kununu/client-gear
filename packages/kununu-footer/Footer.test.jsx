import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line import/no-extraneous-dependencies
import {Link} from 'react-router'; // eslint-disable-line import/no-extraneous-dependencies

import {Footer} from '../kununu-footer';

jest.mock('nukleus');

const infoText = (
  <span>Auf kununu wurden bereits <b className="text-green">1.475.000</b> authentische Erfahrungsberichte über Gehalt, Betriebsklima und Bewerbungsprozesse zu <b className="text-green">297.000</b> Unternehmen abgegeben</span>
);

const footer = (
  <Footer
    infoText={infoText}
    pathname="test"
    tuv
    items={{
      countrySwitcher: [
        {
          icon: <img title="Austrian Flag" alt="Austrian Flag" src="" />,
          link: <Link to={{pathname: '/'}}>test</Link>,
          value: 'Austria',
        },
        {
          icon: <img title="German Flag" alt="German Flag" src="" />,
          link: <Link to={{pathname: '/de'}}>test</Link>,
          value: 'German',
        },
        {
          icon: <img title="Swiss Flag" alt="Swiss Flag" src="" />,
          link: <Link to={{pathname: '/ch'}}>test</Link>,
          value: 'Switzerland',
        },
        {
          icon: <img title="American Flag" alt="American Flag" src="" />,
          link: <Link to={{pathname: '/us'}}>test</Link>,
          value: 'United States',
        },
      ],
      navs: {
        cols: [
          {
            items: [
              {
                link: <Link to="/a">test</Link>,
                value: 'Was ist kununu',
              },
              {
                link: <Link to="/b">test</Link>,
                value: 'Karriere',
              },
              {
                link: <Link to="/c">test</Link>,
                value: 'Presse',
              },
              {
                link: <Link to="/d">test</Link>,
                value: 'News',
              }],
            title: 'Über kununu',
          },
          {
            items: [
              {
                link: <Link to="/e">test</Link>,
                value: 'Gehaltsrechner',
              },
              {
                link: <Link to="/f">test</Link>,
                value: 'Hilfe & Kontakt',
              },
            ],
            title: 'Für Mitarbeiter',
          },
          {
            items: [
              {
                link: <Link to="/g">test</Link>,
                value: 'Unsere Produkte',
              },
              {
                link: <Link to="/h">test</Link>,
                value: 'FAQ',
              },
              {
                link: <Link to="/i">test</Link>,
                value: 'Kontakt',
              },
              {
                link: <Link to="/j">test</Link>,
                value: 'Toolkit',
              },
            ],
            title: 'Für Unternehmen',
          },
        ],
        rows: [
          {
            items: [
              {
                link: <Link to="/k">test</Link>,
                value: 'AGB',
              },
              {
                link: <Link to="/l">test</Link>,
                value: 'Impressum',
              },
              {
                link: <Link to="/m">test</Link>,
                value: 'Datenschutz',
              },
              {
                link: <Link to="/n">test</Link>,
                value: 'Disclaimer',
              },
              {
                link: <Link to="/o">test</Link>,
                value: 'Sitemap',
              },
            ],
          },
        ],
      },
    }}
  />
);

test('Renders Footer without crashing', () => {
  const component = renderer.create(footer);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

