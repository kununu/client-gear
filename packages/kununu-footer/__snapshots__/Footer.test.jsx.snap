exports[`test Renders Footer without crashing 1`] = `
<footer
  className="navbar-default footer"
  id="footer"
  role="banner">
  <div
    className="container-fluid">
    <div
      className="row flex">
      <div
        className="menuColumns visible-xs">
        <div>
          <div>
            <div
              className="h3 text-muted hidden-xs title" />
            <button
              className="visible-xs accordionTitle false"
              onClick={[Function]}>
              <span>
                <span>
                  Austria
                  <span
                    className="titleIcon">
                    <img
                      alt="Austrian Flag"
                      src=""
                      title="Austrian Flag" />
                  </span>
                </span>
              </span>
              <i
                className="fa fa-plus icon" />
            </button>
          </div>
          <ul
            className="
                      nav-sm
                      footerNav
                      col
                      false"
            role="navigation">
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Austria
                  <span
                    className="itemIcon">
                    <img
                      alt="Austrian Flag"
                      src=""
                      title="Austrian Flag" />
                  </span>
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  German
                  <span
                    className="itemIcon">
                    <img
                      alt="German Flag"
                      src=""
                      title="German Flag" />
                  </span>
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Switzerland
                  <span
                    className="itemIcon">
                    <img
                      alt="Swiss Flag"
                      src=""
                      title="Swiss Flag" />
                  </span>
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  United States
                  <span
                    className="itemIcon">
                    <img
                      alt="American Flag"
                      src=""
                      title="American Flag" />
                  </span>
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="menuColumns">
        <div>
          <div>
            <div
              className="h3 text-muted hidden-xs title">
              Über kununu
            </div>
            <button
              className="visible-xs accordionTitle false"
              onClick={[Function]}>
              <span>
                Über kununu
              </span>
              <i
                className="fa fa-plus icon" />
            </button>
          </div>
          <ul
            className="
                      nav-sm
                      footerNav
                      col
                      false"
            role="navigation">
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Was ist kununu
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Karriere
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Presse
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  News
                  
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="menuColumns">
        <div>
          <div>
            <div
              className="h3 text-muted hidden-xs title">
              Für Mitarbeiter
            </div>
            <button
              className="visible-xs accordionTitle false"
              onClick={[Function]}>
              <span>
                Für Mitarbeiter
              </span>
              <i
                className="fa fa-plus icon" />
            </button>
          </div>
          <ul
            className="
                      nav-sm
                      footerNav
                      col
                      false"
            role="navigation">
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Gehaltsrechner
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Hilfe & Kontakt
                  
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="menuColumns">
        <div>
          <div>
            <div
              className="h3 text-muted hidden-xs title">
              Für Unternehmen
            </div>
            <button
              className="visible-xs accordionTitle false"
              onClick={[Function]}>
              <span>
                Für Unternehmen
              </span>
              <i
                className="fa fa-plus icon" />
            </button>
          </div>
          <ul
            className="
                      nav-sm
                      footerNav
                      col
                      false"
            role="navigation">
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Unsere Produkte
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  FAQ
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Kontakt
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Toolkit
                  
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="
                        tuvColumn
                        no-padding
                        hidden-xs
                        hidden-sm
                      ">
        <img
          alt="hi"
          aria-describedby="tooltip"
          className="tuv"
          onBlur={[Function]}
          onClick={null}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          src="" />
      </div>
      <div
        className="infoTextColumn text-right text-center-xs text-center-sm text-muted">
        <div
          className="logoContainer">
          <div
            className="logo light">
            <h1
              className="sr-only">
              kununu GmbH
            </h1>
            <span
              aria-hidden="true"
              className="starSpinner light"
              style={
                Object {
                  "transform": "rotate(0deg)",
                }
              } />
          </div>
        </div>
        <p
          className="infoText">
          <span>
            Auf kununu wurden bereits 
            <b
              className="text-green">
              1.475.000
            </b>
             authentische Erfahrungsberichte über Gehalt, Betriebsklima und Bewerbungsprozesse zu 
            <b
              className="text-green">
              297.000
            </b>
             Unternehmen abgegeben
          </span>
        </p>
        <p
          className="text-xs">
          made with 
          <i
            className="fa fa-heart-o text-red" />
           in Vienna
        </p>
      </div>
    </div>
    <div
      className="bottomMenu">
      <div>
        <div>
          
          <ul
            className="
                      nav-sm
                      footerNav
                      row
                      false"
            role="navigation">
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  AGB
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Impressum
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Datenschutz
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Disclaimer
                  
                </span>
              </a>
            </li>
            <li
              className="footerNavItem false">
              <a
                onClick={[Function]}
                style={Object {}}>
                <span>
                  Sitemap
                  
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="hidden-xs" />
    </div>
  </div>
</footer>
`;
